src/utils/format.utils.ts

```diff
@@ -47,3 +47,165 @@ export const formatMoney = (number: number, options?: FormatMoneyOptions) => {
         return number.toString();
     }
 };
+
+/**
+ * Converts and formats a given date input into a specified string format without usage of external libraries such as moment.js or date-fns.
+ *
+ * @param {Date | number | string} dateInput - The date to be formatted.
+ *   - Can be a Date object, Unix timestamp, or date string.
+ * @param {Intl.DateTimeFormatOptions} [options] - Optional configuration for date formatting.
+ *   - Defaults to { day: "2-digit", month: "2-digit", year: "numeric" }.
+ * @param {string} [format="YYYY-MM-DD"] - The desired output format.
+ *   - Supported formats: 'YYYY-MM-DD', 'DD MMM YYYY', 'MMM DD YYYY'.
+ * @param {boolean} [unix=false] - If true, treats the numeric input as a Unix timestamp.
+ *
+ * @returns {string} A formatted date string according to the specified format.
+ *
+ * @example
+ * // Returns date in 'YYYY-MM-DD' format
+ * getFormattedDateString(new Date('2023-05-15'));
+ * // => "2023-05-15"
+ *
+ * @example
+ * // Returns date in 'DD MMM YYYY' format
+ * getFormattedDateString('2023-05-15', {}, 'DD MMM YYYY');
+ * // => "15 May 2023"
+ */
+export const getFormattedDateString = (
+    dateInput: Date | number | string,
+    options: Intl.DateTimeFormatOptions = { day: "2-digit", month: "2-digit", year: "numeric" },
+    format = "YYYY-MM-DD",
+    unix = false,
+): string => {
+    let dateObj: Date;
+    const dateOptions: Intl.DateTimeFormatOptions = { ...options };
+
+    if (typeof dateInput === "number" && unix) {
+        dateObj = new Date(dateInput * 1000);
+    } else if (typeof dateInput === "string" || dateInput instanceof Date) {
+        dateObj = new Date(dateInput);
```

Comment:
```suggestion
        dateObj = new Date(dateInput);
        if (isNaN(dateObj.getTime())) {
            throw new Error("Invalid date input");
        }
```

example: the dateInput can be `'11:26:33'` which can't be passed to `new Date()`